src/test-apps/happy/test-templates/WeaveInetDNS.py

```diff
@@ -0,0 +1,153 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+#
+#    @file
+#       Implements WeaveInet class that tests Weave Inet Layer among Weave Nodes.
+#
+
+import os
+import sys
+import time
+
+from happy.ReturnMsg import ReturnMsg
+from happy.Utils import *
+from happy.HappyNode import HappyNode
+from happy.HappyNetwork import HappyNetwork
+
+from WeaveTest import WeaveTest
+
+
+# Q: what are the parameters need to specify?
+options = {}
+options["quiet"] = False
+options["node_id"] = None
+options["tap_if"] = None
+options["node_ip"] = None
+options["ipv4_gateway"] = None
+options["dns"] = None
+options["use_lwip"] = False
+
+
+def option():
+    return options.copy()
+
+
+class WeaveInetDNS(HappyNode, HappyNetwork, WeaveTest):
+    
+    def __init__(self, opts = options):
+        HappyNode.__init__(self)
+        HappyNetwork.__init__(self)
+        WeaveTest.__init__(self)
+
+        self.quiet = opts["quiet"]
+        self.node_id = opts["node_id"]
+        self.tap_if = options["tap_if"]
+        self.ipv4_gateway =options["ipv4_gateway"]
+        self.dns = options["dns"]
+        self.use_lwip = options["use_lwip"]
+
+        self.node_process_tag = "WEAVE-INET-NODE"
+
+    def __pre_check(self):
+       # Check if the name of the new node is given
+        if not self.node_id:
+            emsg = "Missing name of the virtual node that should start shell."
+            self.logger.error("[%s] HappyShell: %s" % (self.node_id, emsg))
+            self.exit()
+
+        # Check if virtual node exists
+        if not self._nodeExists():
+            emsg = "virtual node %s does not exist." % (self.node_id)
+            self.logger.error("[%s] HappyShell: %s" % (self.node_id, emsg))
+            self.exit()
+
+    def __gather_results(self):
+        """
+        gather result from get_test_output()
+        """
+        quiet = True
+        results = {}
+
+        results['status'], results['output'] = self.get_test_output(self.node_id, self.node_process_tag, quiet)
+
+        return (results)
+
+
+    def __process_results(self, results):
+        """
+        process results from gather_results()
+        """
+        status = False
+        output = ""
+
+        status = (results['status'] == 0)
+
+        output = results['output']
+
+        return (status, output)
+
+    def __start_node_dnscheck(self):
+        """
+        lwip and socket use different command for now
+        TODO: fix lwip and socket to use similar command future.
```

Comment:
We should get rid of this "TODO" since, as we discussed, the problem is a general one to resolve across **all** of our tests and is not specific to this test.

Reply:
oops, sorry, missed this one:)